src/openssh/patch/0002-Export-remote-info-for-authorization.patch

```diff
@@ -0,0 +1,86 @@
+From 51b3d58afef6796fe0568deb4c3765e24cc828c9 Mon Sep 17 00:00:00 2001
+From: liuh-80 <liuh@microsoft.com>
+Date: Fri, 30 Sep 2022 16:57:03 +0800
+Subject: [PATCH] Export remote info for authorization. authorization.
+
+---
+ auth.c    | 11 +++++++++++
+ auth.h    |  3 +++
+ session.c |  3 +++
+ sshd.c    |  5 +++++
+ 4 files changed, 22 insertions(+)
+
+diff --git a/auth.c b/auth.c
+index c3693ba3f..96d551922 100644
+--- a/auth.c
++++ b/auth.c
+@@ -914,3 +914,14 @@ auth_authorise_keyopts(struct ssh *ssh, struct passwd *pw,
+ 
+ 	return 0;
+ }
++
++/* Export remote IP address and port for authorization. */
++void
++export_remote_info(struct ssh *ssh)
++{
++	const char *remote_ip = ssh_remote_ipaddr(ssh);
++	const int remote_port = ssh_remote_port(ssh);
++	const char remote_addr_port[32 + INET6_ADDRSTRLEN];
```

Comment:
>INET6_ADDRSTRLEN [](http://example.com/codeflow?start=35&length=16)

Do you need reserve one more char for blank?

Reply:
It's not necessary: IP port max value is 65535, so '32 + INET6_ADDRSTRLEN' is enough.